pkg/agent/proxy/types/types.go

```diff
@@ -15,36 +15,72 @@
 package types
 
 import (
+	"net"
+
 	corev1 "k8s.io/api/core/v1"
 	utilnet "k8s.io/utils/net"
 
-	"github.com/vmware-tanzu/antrea/pkg/ovs/openflow"
+	binding "github.com/vmware-tanzu/antrea/pkg/ovs/openflow"
 	k8sproxy "github.com/vmware-tanzu/antrea/third_party/proxy"
 )
 
+type ServiceClient interface {
```

Comment:
While I am fine with defining a separate interface for this, it seems strange to define it under `pkg/agent/proxy` when all non-proxy flows are handled by an interface under `pkg/agent/openflow`
@jianjuns @tnqn

Reply:
Also like to understand the rationale and goals of this change. I do not feel better to let pkg/agent/openflow to depend on pkg/proxy/types.

Reply:
If we go this way, probably ServiceClient should not include any openflow types. It is more like the DP interface to implement Service LB, and openflow client implements it?

Reply:
The rationale for this change is to decouple the DP implementation from the Proxy CP to make it reusable without having to depend on the large struct openflow.client and all dependencies coming from it. Otherwise openflow.client needs to be refactored to be customizable as its pipeline is hardcoded.
It's fine to me that the interface is defined where it's consumed as https://github.com/golang/go/wiki/CodeReviewComments#interfaces suggested (though I think it applies to single consumer case only), which also makes it possible to have another concrete struct implementing the interface without depending on "pkg/agent/openflow".
I agree it would better to keep the type of argument implementation agnostic too, however, the project that reuses the Proxy CP code will also use OVS and the "pkg/ovs/" package. It's just the pkg/agent/openflow is coupled with too many antrea specific logic so it tries to replace that implemenation only. 
@antoninbas @jianjuns Any better idea on this requirement?

Reply:
> It's fine to me that the interface is defined where it's consumed as https://github.com/golang/go/wiki/CodeReviewComments#interfaces suggested (though I think it applies to single consumer case only), which also makes it possible to have another concrete struct implementing the interface without depending on "pkg/agent/openflow".

I am fine with that rationale, but do we intend to further split the Client interface into pieces (Node flows, Pod flows, etc.) and spread these different pieces in different packages? I would say it's a bit surprising to only extract the Service part and move it to the consumer package.

Do you think it's a common / acceptable pattern to have a dedicated package for a related set of Interfaces? We could then split the Client interface into several pieces and place them under a common (e.g. `pkg/agent/datapath`) package.